Gems/Atom/Tools/MaterialEditor/Code/Source/MaterialEditorApplication.cpp

```diff
@@ -139,20 +139,20 @@ namespace MaterialEditor
         AzFramework::AssetSystem::ConnectionSettings connectionSettings;
         AzFramework::AssetSystem::ReadConnectionSettingsFromSettingsRegistry(connectionSettings);
         connectionSettings.m_connectionDirection = AzFramework::AssetSystem::ConnectionSettings::ConnectionDirection::ConnectToAssetProcessor;
-        connectionSettings.m_connectionIdentifier = targetName.c_str();
+        connectionSettings.m_connectionIdentifier = GetBuildTargetName();
         connectionSettings.m_loggingCallback = []([[maybe_unused]] AZStd::string_view logData)
         {
             AZ_TracePrintf("Material Editor", "%.*s", aznumeric_cast<int>(logData.size()), logData.data());
         };
         AzFramework::AssetSystemRequestBus::BroadcastResult(connectedToAssetProcessor,
             &AzFramework::AssetSystemRequestBus::Events::EstablishAssetProcessorConnection, connectionSettings);
 
-        // List of common asset filters for things that need to be compiled to run the material editor
-        // Some of these things will not be necessary once we have proper support for queued asset loading and reloading
-        const AZStd::vector<AZStd::string> assetFiltersArray = { "passes/", "config/", "MaterialEditor/" };
-
         if (connectedToAssetProcessor)
         {
+            // List of common asset filters for things that need to be compiled to run the material editor
+            // Some of these things will not be necessary once we have proper support for queued asset loading and reloading
+            const AZStd::vector<AZStd::string> assetFiltersArray = { "passes/", "config/", "MaterialEditor/" };
+
```

Comment:
If you move this into a virtual function 
AZStd::vector<AZStd::string> AtomToolsApplication::GetCriticalAssetFilters() const
Then you could probably move AssetSystemAvailable to the base class and override GetCriticalAssetFilters to provide the filters.